### PR TITLE
gh-120568: fix file leak in PyUnstable_CopyPerfMapFile

### PR DESCRIPTION
**Pull request by carsonRadtke**
Fixes a file leak in PyUnstable_CopyPerfMapFile found by the Clang Static Analyzer.

* Issue: gh-120568

### CODE REVIEW

File: Python/sysmodule.c [hunk 1]
```diff
@@ -2521,6 +2517,10 @@ PyAPI_FUNC(int) PyUnstable_CopyPerfMapFile(const char* parent_filename) {
     }
     char buf[4096];
     PyThread_acquire_lock(perf_map_state.map_lock, 1);
+    FILE* from = fopen(parent_filename, "r");
```
Comment (sobolevn) [inline Python/sysmodule.c:2518]: There's a behaviour change now: before we opened a file before the thread lock, now it is opened under a lock. What does this imply?

File: Python/sysmodule.c [hunk 2]
```diff
@@ -2521,6 +2517,10 @@ PyAPI_FUNC(int) PyUnstable_CopyPerfMapFile(const char* parent_filename) {
...
+    if (!from) {
```
Comment (kumaraditya303) [inline Python/sysmodule.c:2519]: You forgot to release lock here.

File: Python/sysmodule.c (hunk 1 above)
Comment (kumaraditya303) [reply Python/sysmodule.c:2518]: Doesn't matter much although I would prefer to open it before acquiring lock.

File: Python/sysmodule.c (hunk 2 above)
Comment (carsonRadtke) [reply Python/sysmodule.c:2519]: moved fopen to before PyThread_acquire_lock to satisfy both PR conversations.

File: Python/sysmodule.c (hunk 1 above)
Comment (carsonRadtke) [reply Python/sysmodule.c:2518]: moved fopen to before PyThread_acquire_lock to satisfy both PR conversations.

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**